core/src/main/java/com/github/windsekirun/naraeaudiorecorder/NaraeAudioRecorder.kt

```diff
@@ -192,8 +192,8 @@ class NaraeAudioRecorder {
 
     private fun findRecordWriter() {
         val recordWriter = when (recorderConfig.audioSource) {
-            is DefaultAudioSource -> DefaultRecordWriter(recorderConfig.audioSource)
             is NoiseAudioSource -> NoiseRecordWriter(recorderConfig.audioSource)
```

Comment:
Because `NoiseAudioSource` extends `DefaultAudioSource`, I had to reorder the checks.